app/src/main/java/com/example/criminalintentapp/presentation/AuthenticationViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.criminalintentapp.presentation
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+
+class AuthenticationViewModel : ViewModel() {
+
+    var userRegisterLiveData = MutableLiveData<FirebaseUser>()
+    var userLoginLiveData = MutableLiveData<FirebaseUser>()
+    var failureLiveData = MutableLiveData<String>()
+
+    fun register(emailSignUp: String, passwordSignUp: String) {
+        FirebaseAuth.getInstance().createUserWithEmailAndPassword(
+            emailSignUp,
+            passwordSignUp
+        )
+            .addOnSuccessListener {
```

Comment:
.addOnSuccessListener does not have to be on a new line, it can be placed next to ")". Same for onFailureListener